helper/common/common.go

```diff
@@ -174,3 +174,21 @@ func PadLeftOrTrim(bb []byte, size int) []byte {
 
 	return tmp
 }
+
+// ExtendByteSlice extends given byte slice by needLength parameter.
+// If trim set to true, it puts hard limit on length and capacity to needLength value.
+func ExtendByteSlice(b []byte, needLength int, trim bool) []byte {
```

Comment:
I would prefer that we do two functions for the extension instead, one for the trim and one without or even keep the one of the bitmap as it is. The reason is that `extendByteSlice` (as it is) is a useful function for Go high-performance programming and I do not want to have to worry about misusage.

Reply:
Fixed https://github.com/0xPolygon/polygon-edge/pull/1058/commits/f4fe15458c48131f939419d67a60747946d15d4e